oauth2-authorization-server/src/main/resources/org/springframework/security/oauth2/server/authorization/oauth2-authorization-schema.sql

```diff
@@ -3,23 +3,23 @@ CREATE TABLE oauth2_authorization (
     registered_client_id varchar(100) NOT NULL,
     principal_name varchar(200) NOT NULL,
     authorization_grant_type varchar(100) NOT NULL,
-    attributes varchar(4000) DEFAULT NULL,
+    attributes varchar(12000) DEFAULT NULL,
     state varchar(500) DEFAULT NULL,
-    authorization_code_value blob DEFAULT NULL,
```

Comment:
Let's revert the changes in `oauth2-authorization-schema.sql` and add a new file `oauth2-authorization-schema-postgres.sql` with the updates.

Reply:
@ovidiupopa91 To allow for backwards compatibility, we'll need to support both `blob` and `clob`. What we'll need to do at initialization time (or first call to `JdbcOAuth2AuthorizationService`), is introspect the table metadata to see if `blob` or `clob` is defined. Then we'll know which path to take on the db updates.

Take a look at this Stack Overflow [post](https://stackoverflow.com/questions/16870615/extracting-resultsetmetadata-from-spring-jdbctemplate) on how to retrieve the table metadata. Let me know if you have any questions.

Reply:
hi @jgrandja . Yes, I already did something similar. One question: in my current implementation I'm checking only of the token columns if it's defined as `blob` or `clob`. Do you think that we should check for all token columns? (if there's a combination of `clob` and `blob`)

Reply:
@ovidiupopa91 

> Do you think that we should check for all token columns?

No, this is not necessary. If one of the token columns is a `blob` then we can assume that `oauth2-authorization-schema.sql` was used to create the table. Furthermore, if a token column is a `clob`, then we can assume the new `oauth2-authorization-schema-postgres.sql` was used to create the table.